locals.tf

```diff
@@ -1,10 +1,23 @@
 locals {
-  queue_name          = "${var.customer_name}-RemediationsDLQ.fifo"
-  inbound_queue_name  = "${var.customer_name}-InboundRemediations.fifo"
-  account_id          = data.aws_caller_identity.current.account_id
-  sns_topic_arn       = "arn:aws:sns:${local.region}:${var.organizationID}:${var.snsNotifyTopic}"
-  region              = data.aws_region.current.name
-  role_name           = "${var.customer_name}-RemediationFunction"
-  queue_principal_arn = "arn:aws:iam::${var.organizationID}:root"
-  function_name       = "BridgecrewRemoteRemediation"
+  # Bridgecrew account to receive notifications
+  organization_id                    = "890234264427"
+  sns_notify_topic                   = "handle-customer-actions"
+  queue_name                         = "${var.customer_name}-RemediationsDLQ.fifo"
+  inbound_queue_name                 = "${var.customer_name}-InboundRemediations.fifo"
+  account_id                         = data.aws_caller_identity.current.account_id
+  sns_topic_arn                      = "arn:aws:sns:${local.region}:${local.organization_id}:${local.sns_notify_topic}"
+  region                             = data.aws_region.current.name
+  role_name                          = "${var.customer_name}-RemediationFunction"
+  queue_principal_arn                = "arn:aws:iam::${local.organization_id}:root"
+  function_name                      = "BridgecrewRemoteRemediation"
+  fifo_queue                         = true
+  template_bucket                    = "bc-code-artifacts-${local.organization_id}-"
+  lambda_zip_name                    = "prod/remediations_lambda_c5f16a2212411fd69a5c6a5fe37278617df82f5a.zip"
+  outbound_remediations_endpoint     = "https://dfak3u9wq1.execute-api.us-west-2.amazonaws.com/v1"
+  template_version                   = "0.3.37"
+  batch_size                         = 1
+  runtime                            = "nodejs10.x"
+  lambda_timeout                     = 900
+  maximum_batching_window_in_seconds = 0
+  profile_str                        = var.aws_profile != null ? "--profile ${var.aws_profile}" : ""
```

Comment:
I don't want most of these variables to be documented or editable, which is why I moved them to locals.

How would we update a variable default without updating the module version? Or are you saying that we could just tell the user the correct zip file path to use?

Reply:
If you use variables and default values you can always supply/overide values if the need to arises. If you use locals for non dynamic values you cant, and you might as well hard code the values in the resources

Reply:
in this example - you have set the lambda runtime, if that runtime doesnt exist anymore or you publish a package with a new runtime, then youll have to publish a new module. If it's a default value you can supply in your module.tf declaration and your good.

Reply:
Well, locals still have the obvious value of being reusable and easy to change in one place.

What I want to avoid is users changing things like the runtime on their own and then coming to us when it doesn't work for them. And I want to keep it simple for users.

I think the difference between this module and your typical terraform module is that the goal is not maximum flexibility for covering every use case. It's to create a simple TF equivalent to the CFN stack integration.

Reply:
This module's deployment is relatively predictable and is used by a specific type of consumer (a bridgecrew customer). So it feels ok to me to limit, and by that simplify, the number of configurable parameters.

Reply:
@JamesWoolfenden and I have discussed. a longterm thing would be to create a bridgecrew terraform provider. for now we will continue that way.